src/main/java/webApp/webController.java

```diff
@@ -81,10 +84,13 @@ public SurveyMessage retrieveSurvey(@RequestParam (name="id") int id) {
             for (Question question : survey.get().getQuestions()) {
                 if(question instanceof OpenEndedQuestion) {
```

Comment:
For later refactor, a switch-case statement is far for efficient for cases like theses

Reply:
Good note.

Reply:
Are you sure that a switch statement can be used for instanceof checking?
My reading of these comments suggests it is not https://stackoverflow.com/questions/5579309/is-it-possible-to-use-the-instanceof-operator-in-a-switch-statement

Reply:
Apparently you can do a switch described in here https://stackoverflow.com/questions/36332665/how-to-use-instanceof-in-a-switch-statement.
```
switch (true) {
    case error instanceof NotAllowedError: 
        return res.send(400);

    case error instanceof DatabaseEntryNotFoundError: 
        return res.send(404);

    default:
        log('Failed to do something async with an unspecified error: ', error);
        return res.send(500);
}
```
But has this actually improved anything? The switch true would likely look convoluted to someone unfamiliar with the code.

Reply:
This will be useful to keep in mind for the future.